tests/src/Test/Tasty/Vivado.hs

```diff
@@ -32,12 +33,17 @@ data VivadoTest = VivadoTest
   , entity :: String
   }
 
+createDirIfNotExists :: FilePath -> IO ()
+createDirIfNotExists fp = do
+  b <- doesDirectoryExist fp
+  unless b (createDirectory fp)
+
 instance IsTest VivadoTest where
   run optionSet (VivadoTest dir m top) progressCallback
     | Vivado True <- lookupOption optionSet = do
         hdlDir <- dir
-        createDirectory $ hdlDir </> "ip"
-        createDirectory $ hdlDir </> "project"
+        createDirIfNotExists $ hdlDir </> "ip"
+        createDirIfNotExists $ hdlDir </> "project"
```

Comment:
Yes! I think Peter said it meant our approach wasn't thread-safe. 

I think maybe this can be avoided now that I don't have multiple build targets (because I'm trying to avoid other things)

Reply:
Okay, in that case add an `-- XXX blabla` comment explaining that this is fragile and should be fixed in the future, possibly with an outline on how to fix it.

Reply:
I'm almost sure it is wrong and also it should have been part of the earlier PR, right?

If you run it for multiple build targets, they will all save their project as `project/sim.xpr`. I am certain this will blow up.

Reply:
@martijnbastiaan I've removed the `createDirIfNotExists`, it's just that we need multiple modules. 

I believe we'd need to handle https://github.com/clash-lang/clash-compiler/issues/2264